### PR TITLE
Warn before leaving user edit page

### LINKED ISSUE #1184: Warn before leaving user edit page
inside [src/Pages/UserManager/EditUserInfo.js](https://github.com/SCE-Development/Core-v4/blob/d45184d5de6736561badb46adc1c5121a02bbdd3/src/Pages/UserManager/EditUserInfo.js), if any input was changed, add a prompt to the user "are you sure you want to leave". below is an example on how to do this

https://stackoverflow.com/a/36358770

we should not show the prompt if the user hits cancel, or if the user hits save and then closes the page

### PR DESCRIPTION
**Pull request by weslayer**
adds additional useEffect to EditUserInfo.js in Pages that tracks dataWasChanged, alerts if true.

### CODE REVIEW

**Comment by weslayer** [comment]
Resolves #1184.

**Comment by weslayer** [comment]
Fixed DiscordID not properly being updated for the User.

File: api/main_endpoints/routes/User.js [hunk 1]
```diff
@@ -397,7 +397,6 @@ router.post('/getUserById', async (req, res) => {
     }
 
     const { password, ...omittedPassword } = result._doc;
-
```
Comment (evanugarte) [inline api/main_endpoints/routes/User.js]: undo the change in this file, i.e. add the blank line back

**Review by evanugarte — COMMENTED** [review]
do we need the "original" hooks? i think this can just be

if `dataWasChanged` is true, add the event listener for the prompt

if `dataWasChanged` is false (or becomes false), remove the event listener

File: api/main_endpoints/routes/User.js (hunk 1 above)
Comment (weslayer) [reply api/main_endpoints/routes/User.js]: done

Comment (evanugarte) [reply api/main_endpoints/routes/User.js]: by undoing i would push a commit, and then check in the "files changed" page to make sure `api/main_endpoints/routes/User.js` is not listed in the changes

File: src/Pages/UserManager/EditUserInfo.js [hunk 2]
```diff
@@ -14,19 +14,19 @@ import { sendVerificationEmail } from '../../APIFunctions/Mailer';
...
-  const [email, setEmail] = useState();
+  const [accessLevel, setAccessLevel] = useState('');
+  const [email, setEmail] = useState('');
   const [emailOptIn, setEmailOptIn] = useState();
   const [
     numberOfSemestersToSignUpFor,
     setNumberOfSemestersToSignUpFor
   ] = useState();
-  const [discordId, setDiscordId] = useState();
+  const [discordID, setDiscordID] = useState('');
```
Comment (evanugarte) [inline src/Pages/UserManager/EditUserInfo.js]: do any of these lines even need to be changed?

File: src/Pages/UserManager/EditUserInfo.js [hunk 3]
```diff
@@ -263,7 +279,7 @@ export default function EditUserInfo(props) {
                         checked={!!emailVerified}
                         onChange={(e) => {
                           setDataWasChanged(true);
-                          setEmailVerified(e.target.checked);
+                          setEmailVerified(newEmailVerified);
```
Comment (evanugarte) [inline src/Pages/UserManager/EditUserInfo.js]: `newEmailVerified` is not defined anywhere

File: api/main_endpoints/routes/User.js (hunk 1 above)
Comment (weslayer) [reply api/main_endpoints/routes/User.js]: done for good

File: src/Pages/UserManager/EditUserInfo.js (hunk 2 above)
Comment (weslayer) [reply src/Pages/UserManager/EditUserInfo.js]: done, was just messing around with those values and forgot to change back.

File: src/Pages/UserManager/EditUserInfo.js (hunk 3 above)
Comment (weslayer) [reply src/Pages/UserManager/EditUserInfo.js]: done

File: src/Pages/UserManager/EditUserInfo.js [hunk 4]
```diff
@@ -60,14 +60,30 @@ export default function EditUserInfo(props) {
         setMembershipExpiration(
           new Date(result.responseData.membershipValidUntil)
         );
-        setDiscordId(result.responseData.discordId);
+        setDiscordID(result.responseData.discordID);
```
Comment (evanugarte) [inline src/Pages/UserManager/EditUserInfo.js:63]: undo this change

File: src/Pages/UserManager/EditUserInfo.js [hunk 5]
```diff
@@ -26,7 +26,7 @@ export default function EditUserInfo(props) {
     numberOfSemestersToSignUpFor,
     setNumberOfSemestersToSignUpFor
   ] = useState();
-  const [discordId, setDiscordId] = useState();
+  const [discordID, setDiscordID] = useState('');
```
Comment (evanugarte) [inline src/Pages/UserManager/EditUserInfo.js]: undo pls

**Review by evanugarte — APPROVED** [review]

**Review by syng828 — APPROVED** [review]
looks good

File: src/Pages/UserManager/EditUserInfo.js [hunk 6]
```diff
@@ -81,7 +97,7 @@ export default function EditUserInfo(props) {
       major,
       numberOfSemestersToSignUpFor,
       doorCode,
-      discordID: discordId,
+      discordId: discordId,
```
Comment (jerrylee17) [inline src/Pages/UserManager/EditUserInfo.js]: Unnecessary map since discordId = discordId.

doorCode,
discordId,
pagesPrinted

**Review by jerrylee17 — DISMISSED** [review]

File: src/Pages/UserManager/EditUserInfo.js (hunk 6 above)
Comment (weslayer) [reply src/Pages/UserManager/EditUserInfo.js]: done

File: src/Pages/UserManager/EditUserInfo.js (hunk 5 above)
Comment (weslayer) [reply src/Pages/UserManager/EditUserInfo.js]: done
